astropy/coordinates/name_resolve.py

```diff
@@ -165,9 +165,7 @@ def get_icrs_coordinates(name, parse=False, cache=False):
             domains.append(domain)
 
             # Add the query to the end of the url, add to url list
-            fmt_url = os.path.join(url, "{db}?{name}")
-            fmt_url = fmt_url.format(name=urllib.parse.quote(name), db=db)
-            urls.append(fmt_url)
+            urls.append(urllib.parse.urljoin(url, f"{db}?{urllib.parse.quote(name)}"))
```

Comment:
Is using f-string to build URL secure? I know it is not recommended for logger (https://github.com/python/cpython/issues/90358).

Reply:
I don't know. I *think* loggers are peculiar because they have their string interpolation API, and it doesn't mix well with fstrings. In that regard, I don't see a parallel with `urllib.parse.urljoin`, but maybe I'm missing the point entirely ?
I also don't know if `str.format` is more secure in that regard so ... dealer's call ! Would you rather we preserve this style ?

Reply:
Since @eerovaher is already reviewing, let's see what he says. Thanks!

Reply:
I am not worried about the use of f-strings here. The main security problem (that I am aware of) is that if `name` is something with a very expensive `__str__()` method then calling `SkyCoord.from_name(name)` can be used as a denial-of-service attack. If we are concerned about that then way to mitigate it would be to add an `if not isinsntace(name, str): raise TypeError(...)` check.

Reply:
Well, that's easy enough, I'll push your suggestion. Thanks !

Reply:
When I wrote
> If we are concerned about that...
then the "If" I used was a programmer's if. I don't think we should add the type check in this pull request.